test/Core.Test/Tools/Queries/GetInactiveTwoFactorQueryTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Net;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using Bit.Core.Exceptions;
+using Bit.Core.Settings;
+using Bit.Core.Tools.Queries;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using Microsoft.Extensions.Caching.Distributed;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using Xunit;
+using GlobalSettings = Bit.Core.Settings.GlobalSettings;
+
+namespace Bit.Core.Test.Tools.Queries;
+
+[SutProviderCustomize]
+public class GetInactiveTwoFactorQueryTests
+{
+    public class MockHttpMessageHandler : HttpMessageHandler
+    {
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+            CancellationToken cancellationToken) => Send(request, cancellationToken);
+
+        public virtual Task<HttpResponseMessage> Send(HttpRequestMessage request, CancellationToken token)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task GetInactiveTwoFactor_FromApi_Success(SutProvider<GetInactiveTwoFactorQuery> sutProvider)
```

Comment:
This tests the expected outcome of function calls resulting from a successful call from the API. I disagree that it doesn't test much but do agree there could be an element of testing the custom parser.

Reply:
To expand on my comment a little - you're right that it asserts that `distributedCache.SetAsync` is called with any arguments, and to be fair the use of that dependency is important. But without asserting the values that are passed into it, or the return values of the query, I don't think the unit test provides much assurance that the SUT works as intended, or much protection against regressions.

Also, parsing logic is perfect for unit tests, because it's largely value in -> value out :) 

That's all just my 2 cents though, you're choice whether you want to spend more time on it.

Reply:
@vincentsalucci I see you're picking this back up - unit testing has become more important since my comment above, so I'd really like to see more detailed tests/assertions, and maybe a feature flag to switch between our current solution (hitting the third party server directly) and your new implementation.

Let me know if you want to discuss or pair on it! I still think this is extremely worthwhile to take the load off the third party servers.